internal/vds/metadatahandle.hpp

```diff
@@ -23,7 +27,44 @@ class MetadataHandle {
 
     Axis const& get_axis(Direction const direction) const noexcept (false);
 
-    OpenVDS::IJKCoordinateTransformer coordinate_transformer() const noexcept (true);
+    /**
+     * Extends IJKCoordinateTransformer with ability to translate to openvds
+     * positions - aka system used in Request- functions.
+
+     * We have a problem that OpenVDS' ToVoxelPosition and data request
+     * functions have different definitions of where a datapoint is. E.g. A
+     * transformer will return (0,0,0) for the first sample in the cube. The
+     * request functions on the other hand assume the data is located in the
+     * center of a voxel. I.e. that the first sample is at (0.5, 0.5, 0.5).
+     *
+     * Thus extending IJKCoordinateTransformer with new methods helps us to
+     * bridge this gap.
+     */
+    class CoordinateTransformer : public OpenVDS::IJKCoordinateTransformer{
```

Comment:
Why is it a nested class ?

Reply:
>IJKCoordinateTransformer needs to know about layout which belongs to MetadataHandler.
MetatadataHandler needs to know about CoordinateTransformer and needs to create it.
I want CoordinateTransformer to know about axes, aka it needs to know about MetadataHandler.

This is very I start to think we have a too complicated, entangled and rigid dependency graph between objects